File: mobile/lib/models/manga.dart
// lib/models/manga.dart
class Manga {
  final String id;
  final String title;
  final String? description;
  final String? coverUrl;
  final String? author;
  final String? artist;
  final String? status;
  final String? source;
  final double? score;
  final int? members;
  final int? year;
  final String? contentRating;
  final List<String> tags;

  const Manga({
    required this.id,
    required this.title,
    this.description,
    this.coverUrl,
    this.author,
    this.artist,
    this.status,
    this.source,
    this.score,
    this.members,
    this.year,
    this.contentRating,
    this.tags = const [],
  });

  factory Manga.fromJson(Map<String, dynamic> json) {
    return Manga(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      description: json['description']?.toString(),
      coverUrl: json['cover_url']?.toString(),
      author: json['author']?.toString(),
      artist: json['artist']?.toString(),
      status: json['status']?.toString(),
      source: json['source']?.toString(),
      score: (json['score'] is num) ? (json['score'] as num).toDouble() : double.tryParse(json['score']?.toString() ?? ''),
      members: json['members'] is int ? json['members'] : int.tryParse(json['members']?.toString() ?? ''),
      year: json['year'] is int ? json['year'] : int.tryParse(json['year']?.toString() ?? ''),
      contentRating: json['content_rating']?.toString(),
      tags: (json['tags'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
    );
  }
}

class MangaListResponse {
  final List<Manga> data;
  final int total;

  const MangaListResponse({required this.data, required this.total});

  factory MangaListResponse.fromJson(Map<String, dynamic> json) {
    return MangaListResponse(
      data: (json['data'] as List<dynamic>?)?.map((e) => Manga.fromJson(e)).toList() ?? [],
      total: json['total'] ?? 0,
    );
  }
}

File: backend/app/schemas/manga.py
# backend/app/schemas/manga.py
from pydantic import BaseModel
from typing import Optional


class MangaResponse(BaseModel):
    id: str
    source: str = "mangadex"
    title: str
    description: Optional[str] = None
    status: Optional[str] = None
    year: Optional[int] = None
    content_rating: Optional[str] = None
    tags: list[str] = []
    cover_url: Optional[str] = None
    author: Optional[str] = None
    artist: Optional[str] = None
    mal_id: Optional[int] = None
    score: Optional[float] = None
    members: Optional[int] = None
    cv_id: Optional[int] = None
    issue_count: Optional[int] = None


class MangaListResponse(BaseModel):
    data: list[MangaResponse]
    total: int = 0
    limit: int = 20
    offset: int = 0


class ChapterResponse(BaseModel):
    id: str
    chapter: Optional[str] = None
    title: Optional[str] = None
    volume: Optional[str] = None
    pages: int = 0
    language: str = "en"
    published_at: Optional[str] = None
    scanlation_group: Optional[str] = None
    external_url: Optional[str] = None
    readable: bool = True


class ChapterListResponse(BaseModel):
    data: list[ChapterResponse]
    total: int = 0
    limit: int = 100
    offset: int = 0


class PageListResponse(BaseModel):
    pages: list[str]
    quality: str = "data"


class CharacterResponse(BaseModel):
    mal_id: Optional[int] = None
    name: str
    url: Optional[str] = None
    image_url: Optional[str] = None
    role: str = "Supporting"


class CharacterListResponse(BaseModel):
    data: list[CharacterResponse]
    total: int = 0